mm2src/coins/z_coin/storage/walletdb/wasm/mod.rs

```diff
@@ -0,0 +1,1125 @@
+            .await
+            .is_ok());
+
+        // Account balance should reflect the received note
+        assert_eq!(walletdb.get_balance(AccountId(0)).await.unwrap(), value + value2);
+    }
+
+    #[wasm_bindgen_test]
+    async fn test_scan_cached_blocks_finds_change_notes() {
+        // init blocks_db
+        let ctx = mm_ctx_with_custom_db();
+        let blockdb = BlockDbImpl::new(ctx, TICKER.to_string(), PathBuf::new()).await.unwrap();
+
+        // init walletdb.
+        let walletdb = wallet_db_from_zcoin_builder_for_test(TICKER).await;
+        let consensus_params = consensus_params();
+
+        // Add an account to the wallet
+        let extsk = ExtendedSpendingKey::master(&[]);
+        let extfvk = ExtendedFullViewingKey::from(&extsk);
+        assert!(walletdb.init_accounts_table(&[extfvk.clone()]).await.is_ok());
+
+        // Account balance should be zero
+        assert_eq!(walletdb.get_balance(AccountId(0)).await.unwrap(), Amount::zero());
+
+        // Create a fake compactblock sending value to the address
+        let value = Amount::from_u64(5).unwrap();
+        let (cb1, nf) = fake_compact_block(sapling_activation_height(), BlockHash([0; 32]), extfvk.clone(), value);
+        let cb1_bytes = cb1.write_to_bytes().unwrap();
+        blockdb.insert_block(cb1.height as u32, cb1_bytes).await.unwrap();
+
+        // Scan the cache
+        let scan = DataConnStmtCacheWrapper::new(DataConnStmtCacheWasm(walletdb.clone()));
+        assert!(blockdb
+            .process_blocks_with_mode(consensus_params.clone(), BlockProcessingMode::Scan(scan), None, None)
+            .await
+            .is_ok());
+
+        // Account balance should reflect the received note
+        assert_eq!(walletdb.get_balance(AccountId(0)).await.unwrap(), value);
+
+        // Create a second fake Compactblock spending value from the address
+        let extsk2 = ExtendedSpendingKey::master(&[0]);
+        let to2 = extsk2.default_address().unwrap().1;
+        let value2 = Amount::from_u64(2).unwrap();
+        let cb2 = fake_compact_block_spending(
+            sapling_activation_height() + 1,
+            cb1.hash(),
+            (nf, value),
+            extfvk,
+            to2,
+            value2,
+        );
+        let cb2_bytes = cb2.write_to_bytes().unwrap();
+        blockdb.insert_block(cb2.height as u32, cb2_bytes).await.unwrap();
+
+        // Scan the cache again
+        let scan = DataConnStmtCacheWrapper::new(DataConnStmtCacheWasm(walletdb.clone()));
+        let scan = blockdb
+            .process_blocks_with_mode(consensus_params.clone(), BlockProcessingMode::Scan(scan), None, None)
+            .await;
+        assert!(scan.is_ok());
+
+        // Account balance should equal the change
+        assert_eq!(walletdb.get_balance(AccountId(0)).await.unwrap(), value - value2);
+    }
+
+    fn network() -> Network { Network::TestNetwork }
+
+    //    #[wasm_bindgen_test]
```

Comment:
Why is this commented out?

Reply:
one of the transaction test that takes time to run

Reply:
can you please leave a comment about that so that no one deletes it in the future.